exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRoot.java

```diff
@@ -58,13 +63,170 @@
+        public void run() {
+          try {
+            drillbit.close();
+          } catch (Exception e) {
+            e.printStackTrace();
+          }
+        }
+      }).start();
+      shutdownInfo.put("response", "Shutdown request is triggered");
+      return Response.ok()
+              .entity(shutdownInfo)
+              .header("Access-Control-Allow-Origin", "*")
+              .header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT")
+              .header("Access-Control-Allow-Credentials","true")
+              .allow("OPTIONS").build();
+    }
+    catch (Exception e) {
+      logger.debug("Exception in triggering shutdown request",e);
+      shutdownInfo.put("response", "Error in triggering shutdown request");
+      return Response.ok()
+              .entity(shutdownInfo)
+              .header("Access-Control-Allow-Origin", "*")
+              .header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT")
+              .header("Access-Control-Allow-Credentials","true")
+              .allow("OPTIONS").build();
+    }
+  }
+
+  @SuppressWarnings("resource")
+  @POST
+  @Path("/shutdown")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response ShutdownForcefully() throws Exception {
+    Map<String, String> shutdownInfo = new HashMap<String, String>();
+    try {
+      new Thread(new Runnable() {
+        public void run() {
+          try {
+            drillbit.setForcefulShutdown(true);
+            drillbit.close();
+          } catch (Exception e) {
+            e.printStackTrace();
+          }
+        }
+      }).start();
+      shutdownInfo.put("response", "Forceful shutdown request is triggered");
+      return Response.ok()
+              .entity(shutdownInfo)
+              .header("Access-Control-Allow-Origin", "*")
+              .header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT")
+              .header("Access-Control-Allow-Credentials","true")
+              .allow("OPTIONS").build();
+    }
+    catch (Exception e) {
+      logger.debug("Exception in triggering forceful shutdown request",e);
+      shutdownInfo.put("response", "Error in triggering forceful shutdown");
+      return Response.ok()
+              .entity(shutdownInfo)
+              .header("Access-Control-Allow-Origin", "*")
+              .header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT")
+              .header("Access-Control-Allow-Credentials","true")
+              .allow("OPTIONS").build();
+    }
+  }
+  @SuppressWarnings("resource")
+  @POST
+  @Path("/quiescent")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response drillbitToQuiescentMode() throws Exception {
+    Map<String, String> shutdownInfo = new HashMap<String, String>();
+    try {
+      new Thread(new Runnable() {
+        public void run() {
+          try {
+            drillbit.setQuiescentMode(true);
+            drillbit.close();
+          } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
See above.

This is the third copy of this pattern. Can we have a single function that takes a shutdown mode? Does this mean the `Drillbit` class itself should have such a method:

```
public void shutdown(ShutdownOption option) {
...
```